packages/my-own-react/dom-handlers.js

```diff
@@ -24,24 +26,43 @@ const transformPropToDomProp = (prop) => {
 
 const isNonPrimitiveElement = (element) => typeof element === 'object' && element.type;
 
-const renderComponentElementToHtml = ({ props: { children, ...props }, type }) => {
-  if (typeof type === 'function') {
-    const result = type({ children, ...props });
-    return renderElementToHtml(result);
+const eventHandlersProps = ['onClick', 'onChange', 'onSubmit'];
+
+const addEventHandler = (domElement, { key, value }) => {
+  switch (key) {
+    case 'onClick':
+      return domElement.addEventListener('click', value);
+    case 'onChange':
+      return domElement.addEventListener('input', value);
```

Comment:
```suggestion
      // for the `change` event to trigger, the user is required to leave the field and come back
      // so it seems like React decided to use the `input` event under the hood
      return domElement.addEventListener('input', value);
```

And maybe we can/should confirm that from the docs?

Reply:
Nothing in the docs as far as I could tell, but the code seems to indicate something along those lines [here](https://github.com/facebook/react/blob/8e2bde6f2751aa6335f3cef488c05c3ea08e074a/packages/react-dom-bindings/src/events/plugins/ChangeEventPlugin.js#L45)